dashboard/app/models/levels/level_group.rb

```diff
@@ -240,6 +240,60 @@ def self.get_survey_results(script, section)
     end.compact
   end
 
+  # Surveys: Returns all anonymous survey results, given a script and a section.
+  #
+  # The results look like this.  For each level_group_id, levelgroup_results is an array
+  # wih an entry per contained sublevel.  Inside each entry is an array of results
+  # which has been shuffled to increase student anonymity.  There is an entry for each
+  # student who has submitted the overall LevelGroup, whether they have given a valid
+  # answer to that sublevel question or not, which explains the empty hashes
+  # intermingled with real results.
+  # [ 23432:
+  #   { stage_name: "Stage 30: Anonymous student survey",
+  #     levelgroup_results: [
+  #       {
+  #         type: "multi",
+  #         question: "Computer science is fun",
+  #         results: [
+  #           {answer_index: 0},
+  #           {},
+  #           {answer_index: 0}],
+  #         answer_texts: ["Agree", "Disagree", "Not sure"]}},
+  #       {
+  #         type: "free_response",
+  #         question: "Why are you doing this class?  Give at least two reasons.",
+  #         results: [
+  #           {result: ""},
+  #           {result: "I like making games, and I also like the lifestyle."},
+  #           {}]}]}]
+  def self.get_summarized_survey_results(script, section)
+    level_group_script_levels = script.script_levels.includes(:levels).where('levels.type' => 'LevelGroup')
```

Comment:
This kicks off a DB query, you can compute this array from the script cache instead, using something like:
```
script.script_levels.select {|sl| sl.levels.first.is_a? LevelGroup}
```
You might also check `long_assessment?` and `anonymous?` in the select block instead of having the `next unless` lines below.

Reply:
Also, for future reference: To check if this kicks off a query in the rails console do this:
1. set up the "cache" `Script.with_associated_models.find_by_name 'csp3-2018'`
2. run the command and see if the output gives us a query or just the answer